libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2404,6 +2503,13 @@ MemberAccess const* SMTEncoder::isEmptyPush(Expression const& _expr) const
 	return nullptr;
 }
 
+bool SMTEncoder::isPublicGetter(Expression const* _expr) {
+	if (!isTrustedExternalCall(_expr))
+		return false;
+	auto varDecl = dynamic_cast<VariableDeclaration const*>((dynamic_cast<MemberAccess const*>(_expr))->annotation().referencedDeclaration);
```

Comment:
```suggestion
	auto varDecl = dynamic_cast<VariableDeclaration const*>(
	    dynamic_cast<MemberAccess const*>(_expr)->annotation().referencedDeclaration
	);
```

Reply:
Actually, we first need to check if `_expr` is a `MemberAccess`. I guess now that's guaranteed by `isTrustedExternalCall`, right?

Reply:
Yes, this is now checked in `isTrustedExternalCall`. Should we duplicate the check here?